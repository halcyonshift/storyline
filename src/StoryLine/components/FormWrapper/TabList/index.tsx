import { SyntheticEvent } from 'react'
import { TabList as MuiTabList } from '@mui/lab'
import { Badge, Tab } from '@mui/material'
import { useTranslation } from 'react-i18next'
import { FormTabListProps } from '../types'

const TabList = ({ setValue, tabList, errorBadges, notes }: FormTabListProps) => {
    const { t } = useTranslation()

    return (
        <MuiTabList onChange={(_: SyntheticEvent, value: string) => setValue(value)} aria-label=''>
            {tabList.map((tab, index) => (
                <Tab
                    key={`tab-${index + 1}`}
                    label={
                        errorBadges[index.toString()] ? (
                            <Badge variant='dot' color='error'>
                                {tab}
                            </Badge>
                        ) : (
                            tab
                        )
                    }
                    value={(index + 1).toString()}
                />
            ))}
            {notes.filter((note) => note.image).length ? (
                <Tab label={t('component.formWrapper.tab.images')} value='images' />
            ) : null}
            {notes.length ? (
                <Tab label={t('component.formWrapper.tab.notes')} value='notes' />
            ) : null}
        </MuiTabList>
    )
}

export default TabList
