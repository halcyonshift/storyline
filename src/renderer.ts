import { init } from '@sentry/electron/renderer'
import { init as reactInit } from '@sentry/react'

if (parseInt(process.env.DEBUG) === 0) {
    init({ dsn: process.env.SENTRY }, reactInit)
}

import '@fontsource/roboto/300.css'
import '@fontsource/roboto/400.css'
import '@fontsource/roboto/500.css'
import '@fontsource/roboto/700.css'
import './index.css'
import './StoryLine'
