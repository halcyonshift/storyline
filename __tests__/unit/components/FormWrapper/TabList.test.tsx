import { TabContext } from '@mui/lab'
import userEvent from '@testing-library/user-event'
import renderer from 'react-test-renderer'
import TabList from '@sl/components/FormWrapper/TabList'
import { render, screen } from '../../test-utils'

const mockSetValue = jest.fn()

describe('<TabList />', () => {
    const tabList = (
        <TabContext value='1'>
            <TabList
                setValue={mockSetValue}
                tabList={['General', 'Other']}
                errorBadges={{}}
                notes={[]}
            />
        </TabContext>
    )

    beforeAll(() => {
        render(tabList)
    })

    it('calls setValue on tab press', async () => {
        await userEvent.click(screen.getAllByRole('tab')[1])
        expect(mockSetValue).toBeCalledWith('2')
    })

    it('shows error badges', async () => {
        const { container } = render(
            <TabContext value='1'>
                <TabList
                    setValue={mockSetValue}
                    tabList={['General', 'Other']}
                    errorBadges={{ '1': 2 }}
                    notes={[]}
                />
            </TabContext>
        )
        expect(container.getElementsByClassName('MuiBadge-colorError').length).toBe(1)
    })

    it('renders correctly', () => {
        const tree = renderer.create(tabList).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
