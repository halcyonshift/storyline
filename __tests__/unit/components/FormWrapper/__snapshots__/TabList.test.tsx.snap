// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabList /> renders correctly 1`] = `
<div
  className="MuiTabs-root css-1ujnqem-MuiTabs-root"
>
  <div
    className="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
    onScroll={[Function]}
    style={
      {
        "marginBottom": -0,
        "overflow": "hidden",
      }
    }
  >
    <div
      aria-label=""
      aria-orientation={null}
      className="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
      onKeyDown={[Function]}
      role="tablist"
    >
      <button
        aria-controls={null}
        aria-selected={true}
        className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
        disabled={false}
        id={null}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="tab"
        tabIndex={0}
        type="button"
      >
        General
        <span
          className="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
          style={{}}
        />
      </button>
      <button
        aria-controls={null}
        aria-selected={false}
        className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
        disabled={false}
        id={null}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        Other
      </button>
    </div>
  </div>
</div>
`;
